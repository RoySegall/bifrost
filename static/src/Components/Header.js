import React from 'react';
import {Configuration, Logout} from '../Fonts'

function Header() {
  return <header>
    <div className="container grid grid-cols-2 gap-6 pt-2 pb-3">

      <div>
        <h1 className="text-4xl">Welcome, dummy</h1>
      </div>

      <div>
        <div className="flex flex-row justify-center">
          <div className="text-2xl">{Configuration()} <a href="/settings">Settings</a></div>
          <div className="text-2xl ml-4">{Logout()} <a href="auth/logout">Logout</a></div>
        </div>
      </div>

    </div>

    <hr className="bg-red-500"/>
  </header>;
}

export default Header;
