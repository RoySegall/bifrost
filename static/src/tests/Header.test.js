import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Components/Header';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('Testing the header component', () => {

    const component = renderer.create(<Header />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Testing the logout button', () => {
    // Set a generic access token.
    localStorage.setItem('token', '🍕');

    const header = mount(<Header />);

    expect(localStorage.getItem('token')).toBe('🍕');

    header.find('#logout').simulate('click');

    expect(localStorage.getItem('token')).toBe(null);
});
