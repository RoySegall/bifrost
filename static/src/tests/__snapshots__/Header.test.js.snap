// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the header component 1`] = `
<header
  className="header"
>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h1>
        Welcome, dummy
      </h1>
    </div>
    <div
      className="col-6 menu text-right"
    >
      <ul
        className="list-inline"
      >
        <li
          className="list-inline-item"
        >
          <i
            className="fal fa-user-cog"
          />
           
          <a
            href="/settings"
          >
            Settings
          </a>
        </li>
        |
        <li
          className="list-inline-item"
        >
          <i
            className="fal fa-sign-out"
          />
           
          <a
            href="auth/logout"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </div>
  <hr />
</header>
`;
